### PR TITLE
add module for checking crc checksums and log crc check failues

### PR DESCRIPTION
**Pull request by willus10245**
This adds some data verification by checking the crc check sums sent with the data and verifying that the data wasn't corrupted in transit.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -31,6 +31,7 @@ defmodule SGP30.MixProject do
   defp deps do
     [
       {:circuits_i2c, "~> 0.3"},
+      {:crc, "~> 0.10"},
```
Comment (jjcarstens) [inline mix.exs]: How would you feel about keeping as much native as possible and using `:cerlc` here instead? Or, maybe you have already gone through that decision and this was the result?

File: lib/sgp30/crc.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
+defmodule SGP30.CRC do
+  def check(data_crc_tuples) when is_list(data_crc_tuples) do
+    data_crc_tuples
+    |> Enum.all?(&check/1)
+  end
+
+  def check({data, crc}) do
+    data
+    |> calculate()
+    |> Kernel.==(crc)
```
Comment (jjcarstens) [inline lib/sgp30/crc.ex]:
```suggestion
    calculate(data) == crc
```
This is a stylistic nit - I'm not a huge fan of piping to Kernel for simple calls, if you're open to changing it

**Review by jjcarstens — APPROVED** [review]
thanks @willus10245 ! ❤️ Looks great!

Just a couple style nits, but love the functionality added

File: mix.exs (hunk 1 above)
Comment (willus10245) [reply mix.exs]: Yeah, I hadn't seen that library! I'll make that swap

File: lib/sgp30/crc.ex (hunk 2 above)
Comment (willus10245) [reply lib/sgp30/crc.ex]: No problem at all.

File: mix.exs (hunk 1 above)
Comment (willus10245) [reply mix.exs]: @jjcarstens done!

**Pull request merged**